rasa/shared/core/domain.py

```diff
@@ -699,23 +700,41 @@ def num_states(self):
 
     @rasa.shared.utils.common.lazy_property
     def retrieval_intent_templates(self) -> Dict[Text, List[Dict[Text, Any]]]:
-        """Return only the templates which are defined for retrieval intents"""
+        """Return only the responses which are defined for retrieval intents."""
+        rasa.shared.utils.io.raise_deprecation_warning(
+            "The terminology 'template' is deprecated and replaced by 'response'. This will be removed in the future."
```

Comment:
let's specify the specific version 
```suggestion
            "The terminology 'template' is deprecated and replaced by 'response'. This will be removed in Rasa Open Source 3.0.0."
```